### PR TITLE
FOUNDATIONS:  `Time` Retrieve All

### PR DESCRIPTION
**Pull request by Ulugbek4499**
Closes #257

### CODE REVIEW

File: Tarteeb.Api.Tests.Unit/Services/Foundations/TimeSlots/TimeServiceTests.Logic.RetrieveAll.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        public void ShouldRetrieveAllTimes()
+        {
+            //given
+            IQueryable<Time> randomTimes = CreateRandomTimes();
+            IQueryable<Time> storageTimes = randomTimes;
+            IQueryable<Time> expectedTimes = storageTimes;
+
+            this.storageBrokerMock.Setup(broker =>
+                broker.SelectAllTimes()).Returns(storageTimes);
+            //when
```
Comment (GulchexrasGithub) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/TimeSlots/TimeServiceTests.Logic.RetrieveAll.cs:26]: Empty line required

File: Tarteeb.Api.Tests.Unit/Services/Foundations/TimeSlots/TimeServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+        private IQueryable<Time> CreateRandomTimes()
+        {
+            return CreateTimeFiller(dates: GetRandomDateTime())
+                .Create(count: GetRandomNumber()).AsQueryable();
+        }
+
+        private static int GetRandomNumber() =>
+             new IntRange(min: 2, max: 99).GetValue();
+        private static string GetRandomString() =>
```
Comment (GulchexrasGithub) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/TimeSlots/TimeServiceTests.cs:58]: Empty line required

**Review by GulchexrasGithub — CHANGES_REQUESTED** [review]
Retrieve All Timega oid bo'lmagan classlarni tozalamang.

File: Tarteeb.Api.Tests.Unit/Services/Foundations/TimeSlots/TimeServiceTests.Logic.RetrieveAll.cs (hunk 1 above)
Comment (Ulugbek4499) [reply Tarteeb.Api.Tests.Unit/Services/Foundations/TimeSlots/TimeServiceTests.Logic.RetrieveAll.cs:26]: Resolved

File: Tarteeb.Api/Services/Foundations/Times/TimeService.Exceptions.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+        private TimeDependencyException CreateAndLogCriticalDependencyException(Xeption exception)
+        {
+            var timeDependencyException = new TimeDependencyException(exception);
+            this.loggingBroker.LogCritical(timeDependencyException);
+
+            return timeDependencyException;
+        }
+
+        private TimeServiceException CreateAndLogServiceException(Exception exception)
```
[the remark this answers is not in the document]
Comment (Ulugbek4499) [reply Tarteeb.Api/Services/Foundations/Times/TimeService.Exceptions.cs:47]: Resolved

File: Tarteeb.Api.Tests.Unit/Services/Foundations/TimeSlots/TimeServiceTests.cs (hunk 2 above)
Comment (Ulugbek4499) [reply Tarteeb.Api.Tests.Unit/Services/Foundations/TimeSlots/TimeServiceTests.cs:58]: resolved

**Comment by Ulugbek4499** [comment]
> Retrieve All Timega oid bo'lmagan classlarni tozalamang.

Please give more info about it.

**Comment by GulchexrasGithub** [comment]
> > Retrieve All Timega oid bo'lmagan classlarni tozalamang.
> 
> Please give more info about it.

Don't clean files you did'nt work on, for example you have cleaned 

> > Retrieve All Timega oid bo'lmagan classlarni tozalamang.
> 
> Please give more info about it.

Don't clean files you did'nt work on, for example you have cleaned 
`Tarteeb.Api/Services/Orchestrations/UserSecurityOrchestrationService.Exceptions.cs`
`Tarteeb.Api/Models/Orchestrations/UserTokens/Exceptions/FailedUserTokenOrchestrationException.cs`
`Tarteeb.Api/Controllers/TicketsController.cs`
`Tarteeb.Api/Brokers/Storages/StorageBroker.Users.cs`
`Tarteeb.Api.Tests.Unit/Services/Foundations/Securities/SecurityServiceTests.cs` and others, but you didn't work on this files.  Use `Ctrl k e` for clean.

**Review by GulchexrasGithub — APPROVED** [review]

**Comment by Ulugbek4499** [comment]
> > > Retrieve All Timega oid bo'lmagan classlarni tozalamang.
> > 
> > 
> > Please give more info about it.
> 
> Don't clean files you did'nt work on, for example you have cleaned
> 
> > > Retrieve All Timega oid bo'lmagan classlarni tozalamang.
> > 
> > 
> > Please give more info about it.
> 
> Don't clean files you did'nt work on, for example you have cleaned `Tarteeb.Api/Services/Orchestrations/UserSecurityOrchestrationService.Exceptions.cs` `Tarteeb.Api/Models/Orchestrations/UserTokens/Exceptions/FailedUserTokenOrchestrationException.cs` `Tarteeb.Api/Controllers/TicketsController.cs` `Tarteeb.Api/Brokers/Storages/StorageBroker.Users.cs` `Tarteeb.Api.Tests.Unit/Services/Foundations/Securities/SecurityServiceTests.cs` and others, but you didn't work on this files. Use `Ctrl k e` for clean.

Thanks, I got it!

**Pull request merged**